### PR TITLE
DPE-2653 profile limit option

### PR DESCRIPTION
**Pull request by paulomach**
Port of vm [PR#331](https://github.com/canonical/mysql-operator/pull/331)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ boto3 = "^1.28.22"
 ops = ">=2.0.0"
 # data_platform_libs/v0/upgrade.py
 poetry-core = "*"
-pydantic = "^1.10"
+pydantic = "^1.10, <2"
```
Comment (shayancanonical) [inline pyproject.toml:23]: is the <2 needed? i thought ^1.10 will upgrade to a new version as long as the major version is `1`

**Review by shayancanonical — APPROVED** [review]

**Review by taurus-forever — DISMISSED** [review]
Thank you!

File: pyproject.toml (hunk 1 above)
Comment (paulomach) [reply pyproject.toml:25]: 056c5f8

[the remark this answers is not in the document]
Comment (paulomach) [reply pyproject.toml:25]: 056c5f8

File: src/config.py [hunk 2]
```diff
@@ -92,3 +94,24 @@ def profile_limit_memory_validator(cls, value: int) -> Optional[int]:
...
+        if len(value) > 32:
+            raise ValueError("User name constrained to 32 characters")
+
+        return value
+
+    @validator("mysql_interface_database", "mysql_root_interface_database")
+    @classmethod
+    def database_name_validator(cls, value: str) -> Optional[str]:
+        """Check database name is valid."""
+        if not re.match(r"^[^\\\/?%*:|\"<>.]{1,64}$", value):
```
[the remark this answers is not in the document]
Comment (paulomach) [reply src/config.py:111]: not really, the pattern is constrained by `^` and `$`

you can test it with:
```python
import re
str_64 = 64*"a"
str_65 = 65*"b"
pattern = r"^[^\\\/?%*:|\"<>.]{1,64}$"
if re.match(pattern, str_64):
    print("str_64 match")
if re.fullmatch(pattern, str_64):
    print("str_64 fullmatch")
if re.match(pattern, str_65):
    print("str_65 match")
if re.fullmatch(pattern, str_65):
    print("str_65 fullmatch")
```

**Review by taurus-forever — APPROVED** [review]
Tnx!

**Pull request merged**